### PR TITLE
Fix print string in radiff -qu

### LINKED ISSUE #16190: Do not print strings in `radiff2 -qu file1 file2`
`radiff2 -qu file1 file2` currently prints strings, while it's unnecessary. For now the only difference between `radiff2 -qu` and `radiff2 -u` is the color. Would be nice to have the option to print just changes, without the strings or change the behavior of the current one.

```
-0x000004f8:05 "\x05"
-0x000004fc:07 "\a"
-0x00000500:08 "\b"
-0x00000508:1800 "\x18"
+0x00000508:3a03
-0x00000510:09 "\t"
+0x00000510:34
-0x00000514:03 "\x03"
+0x00000520:01
-0x00000528:4803 "H\x03"
```
should become
```
-0x000004f8:05
-0x000004fc:07
-0x00000500:08
-0x00000508:1800
+0x00000508:3a03
-0x00000510:09
+0x00000510:34
-0x00000514:03
+0x00000520:01
-0x00000528:4803
```
Also I am not sure - may be it's better to align the data bytes, add spacing between them like in the hex mode?
```
-0x000004f8: 05
-0x000004fc: 07
-0x00000500: 08
-0x00000508: 18 00
+0x00000508: 3a 03
-0x00000510: 09
+0x00000510: 34
-0x00000514: 03
+0x00000520: 01
-0x00000528: 48 03
```

### PR DESCRIPTION
**Pull request by kuqadk3**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

-qu options still print out strings, this fix makes it not print string out.

**Test plan**

...

**Closing issues**

closes #16190
...

### CODE REVIEW

File: test/new/db/tools/radiff2 [hunk 1]
```diff
@@ -50,8 +50,8 @@ NAME=radiff2 unified string comparison
 FILE=-
 CMDS=!!radiff2 -quz ../bins/elf/elf_one_symbol_shdr ../bins/elf/elf_one_symbol_shdr1
 EXPECT=<<EOF
--0x00000000:48656c6c "Hello world!\n"
-+0x00000000:41414141
+-0x00000000:48 65 6c 6c
```
Comment (radare) [inline test/new/db/tools/radiff2]: This output looks more like a regression than an improvement

Comment (kuqadk3) [reply test/new/db/tools/radiff2]: The quiet mode supposed not to print string out, isn't it?

**Comment by XVilka** [comment]
@radare it's not a regression - see https://github.com/radareorg/radare2/issues/16190

**Comment by kuqadk3** [comment]
Btw, look at #561

https://github.com/radareorg/radare2/issues/561#issuecomment-379661261

File: libr/main/radiff2.c [hunk 2]
```diff
@@ -140,14 +140,15 @@ static int cb(RDiff *d, void *user, RDiffOp *op) {
 				printf ("-0x%08"PFMT64x":", op->a_off);
 				int len = op->a_len; // R_MIN (op->a_len, strlen (op->a_buf));
 				for (i = 0; i < len; i++) {
-					printf ("%02x", op->a_buf[i]);
+					printf ("%02x ", op->a_buf[i]);
```
Comment (radare) [inline libr/main/radiff2.c]: dont put spaces between hex numbers. this is an hexpair string

File: libr/main/radiff2.c [hunk 3]
```diff
@@ -140,14 +140,15 @@ static int cb(RDiff *d, void *user, RDiffOp *op) {
...
 				}
-				char *p = r_str_escape ((const char*)op->a_buf);
-				printf (" \"%s\"\n", p);
-				free (p);
 				if (!quiet) {
+					char *p = r_str_escape ((const char*)op->a_buf);
+					printf (" \"%s\"", p);
```
Comment (radare) [inline libr/main/radiff2.c]: maybe worth to chk if  is not null

File: test/new/db/tools/radiff2 (hunk 1 above)
Comment (radare) [reply test/new/db/tools/radiff2]: the string is an useful  hint imho

File: libr/main/radiff2.c (hunk 2 above)
Comment (kuqadk3) [reply libr/main/radiff2.c]: Yeah, but I think it's easy to look at if we separate each byte with space.
Especially with long hex string, it's really hard to see.

File: test/new/db/tools/radiff2 (hunk 1 above)
Comment (kuqadk3) [reply test/new/db/tools/radiff2]: Yeah, but for the first program, the string was hidden in the quiet mode which it's not for the second program. Which doesn't make sense, right?

Btw if user wants strings, they can use "-u" option though.

File: libr/main/radiff2.c (hunk 3 above)
Comment (kuqadk3) [reply libr/main/radiff2.c]: There is a check before that already

https://github.com/radareorg/radare2/blob/778c42cfa7b9783b39aeda52993fa6b4fc7161f1/libr/main/radiff2.c#L154-L155

**Pull request merged**